dns.go

```diff
@@ -65,24 +66,36 @@ func (d *DNSServer) Start() {
 					addr, len(dnsMsg.Question))
 				return
 			}
-			domainName := strings.ToLower(dnsMsg.Question[0].Name)
-			ff := strings.LastIndex(domainName, d.hostname)
-			if ff < 0 {
-				log.Printf("invalid name: %s",
+
+			if !dns.IsSubDomain(d.hostname, dnsMsg.Question[0].Name) {
+				log.Printf("%s: invalid hostname: %v", addr,
 					dnsMsg.Question[0].Name)
 				return
 			}
-
-			wantedSF := wire.SFNodeNetwork
-			labels := dns.SplitDomainName(domainName)
-			if labels[0][0] == 'x' && len(labels[0]) > 1 {
-				wantedSFStr := labels[0][1:]
+			var wantedSF wire.ServiceFlag
+			numLabels := dns.CountLabel(dnsMsg.Question[0].Name) - minRequiredLabels
+			switch numLabels {
+			case 0:
+				wantedSF = wire.SFNodeNetwork
+			case 1:
+				labels := dns.SplitDomainName(dnsMsg.Question[0].Name)
+				label := strings.ToLower(labels[0])
+				if len(label) < 2 || label[0] != 'x' {
```

Comment:
So the idea is the query can be for a hostname like "x4.mainnet-seed.dcrdata.org" to look for nodes with service flag = 4 (SFNodeCF)?  Just a brief comment to this extent would be nice.